Adopt more smart pointers in PluginView.cpp

cdumez opened the pull request:
#### 2c6e63feae97417bfd99cafef71fbcd42a0241eb
<pre>
Adopt more smart pointers in PluginView.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=269502">https://bugs.webkit.org/show_bug.cgi?id=269502</a>

Reviewed by Tim Horton.

* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::protectedParent const):
* Source/WebCore/platform/Widget.h:
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::Stream::start):
(WebKit::PluginView::Stream::cancel):
(WebKit::PluginView::Stream::didReceiveResponse):
(WebKit::PluginView::Stream::didReceiveData):
(WebKit::PluginView::Stream::didFail):
(WebKit::PluginView::Stream::didFinishLoading):
(WebKit::PluginView::create):
(WebKit::PluginView::PluginView):
(WebKit::PluginView::~PluginView):
(WebKit::PluginView::webPage const):
(WebKit::PluginView::manualLoadDidReceiveResponse):
(WebKit::PluginView::manualLoadDidReceiveData):
(WebKit::PluginView::manualLoadDidFinishLoading):
(WebKit::PluginView::protectedPluginElement const):
(WebKit::PluginView::layerHostingStrategyDidChange):
(WebKit::PluginView::manualLoadDidFail):
(WebKit::PluginView::didBeginMagnificationGesture):
(WebKit::PluginView::didEndMagnificationGesture):
(WebKit::PluginView::setPageScaleFactor):
(WebKit::PluginView::pageScaleFactor const):
(WebKit::PluginView::webPageDestroyed):
(WebKit::PluginView::setDeviceScaleFactor):
(WebKit::PluginView::accessibilityAssociatedPluginParentForElement const):
(WebKit::PluginView::accessibilityObject const):
(WebKit::PluginView::initializePlugin):
(WebKit::PluginView::layerHostingStrategy const):
(WebKit::PluginView::platformLayer const):
(WebKit::PluginView::graphicsLayer const):
(WebKit::PluginView::scroll):
(WebKit::PluginView::scrollPositionForTesting const):
(WebKit::PluginView::horizontalScrollbar):
(WebKit::PluginView::verticalScrollbar):
(WebKit::PluginView::wantsWheelEvents):
(WebKit::PluginView::paint):
(WebKit::PluginView::countFindMatches):
(WebKit::PluginView::findString):
(WebKit::PluginView::selectionString const):
(WebKit::PluginView::handleEvent):
(WebKit::PluginView::handleEditingCommand):
(WebKit::PluginView::isEditingCommandEnabled):
(WebKit::PluginView::willDetachRenderer):
(WebKit::PluginView::usesAsyncScrolling const):
(WebKit::PluginView::scrollingNodeID const):
(WebKit::PluginView::didAttachScrollingNode):
(WebKit::PluginView::liveResourceData const):
(WebKit::PluginView::performDictionaryLookupAtLocation):
(WebKit::PluginView::notifyWidget):
(WebKit::PluginView::viewGeometryDidChange):
(WebKit::PluginView::viewVisibilityDidChange):
(WebKit::PluginView::clipRectInWindowCoordinates const):
(WebKit::PluginView::focusPluginElement):
(WebKit::PluginView::pendingResourceRequestTimerFired):
(WebKit::PluginView::shouldCreateTransientPaintingSnapshot const):
(WebKit::PluginView::isBeingDestroyed const):
(WebKit::PluginView::pdfDocumentForPrinting const):
(WebKit::PluginView::pdfDocumentSizeForPrinting const):
(WebKit::PluginView::accessibilityHitTest const):
(WebKit::PluginView::lookupTextAtLocation const):
(WebKit::PluginView::rectForSelectionInRootView const):
(WebKit::PluginView::contentScaleFactor const):
(WebKit::PluginView::isUsingUISideCompositing const):
(WebKit::PluginView::didChangeSettings):
(WebKit::PluginView::windowActivityDidChange):
(WebKit::PluginView::didSameDocumentNavigationForFrame):
* Source/WebKit/WebProcess/Plugins/PluginView.h:

Canonical link: <a href="https://commits.webkit.org/274794@main">https://commits.webkit.org/274794@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/92aadd494e2ed9938bc455fd03f50b27ce3ef354

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40065 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19077 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42441 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42610 "Failed to checkout and rebase branch from PR 24544") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/36049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42372 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21986 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16406 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/42610 "Failed to checkout and rebase branch from PR 24544") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40639 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16061 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/34608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/42610 "Failed to checkout and rebase branch from PR 24544") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13923 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/35560 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43888 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/36349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/35896 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39654 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/12213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/16499 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8988 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/16548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/16143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PluginView.cpp:
```diff
@@ -830,11 +844,12 @@ void PluginView::pendingResourceRequestTimerFired()
     ASSERT(!m_stream);
-    m_stream = PluginView::Stream::create(this, *request);
-    m_stream->start();
+    Ref stream = PluginView::Stream::create(this, *request);
+    m_stream = stream.copyRef();
```
Could we not do `m_stream = PluginView::Stream::create(this, *request);`? I assumed that would invoke `template<typename X> RefPtr& operator=(Ref<X>&&);`

cdumez replied:
The issue wasn't that assignment. The issue is that I need a `Ref` *on the stack* so I can call `stream->start();` safely on the next line.

aprotyas replied:
oh! Okay, cool

aprotyas reviewed (commented):
LGTM with one question

cdumez committed:
Adopt more smart pointers in PluginView.cpp
https://bugs.webkit.org/show_bug.cgi?id=269502

Reviewed by Tim Horton.

* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::protectedParent const):
* Source/WebCore/platform/Widget.h:
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::Stream::start):
(WebKit::PluginView::Stream::cancel):
(WebKit::PluginView::Stream::didReceiveResponse):
(WebKit::PluginView::Stream::didReceiveData):
(WebKit::PluginView::Stream::didFail):
(WebKit::PluginView::Stream::didFinishLoading):
(WebKit::PluginView::create):
(WebKit::PluginView::PluginView):
(WebKit::PluginView::~PluginView):
(WebKit::PluginView::webPage const):
(WebKit::PluginView::manualLoadDidReceiveResponse):
(WebKit::PluginView::manualLoadDidReceiveData):
(WebKit::PluginView::manualLoadDidFinishLoading):
(WebKit::PluginView::protectedPluginElement const):
(WebKit::PluginView::layerHostingStrategyDidChange):
(WebKit::PluginView::manualLoadDidFail):
(WebKit::PluginView::didBeginMagnificationGesture):
(WebKit::PluginView::didEndMagnificationGesture):
(WebKit::PluginView::setPageScaleFactor):
(WebKit::PluginView::pageScaleFactor const):
(WebKit::PluginView::webPageDestroyed):
(WebKit::PluginView::setDeviceScaleFactor):
(WebKit::PluginView::accessibilityAssociatedPluginParentForElement const):
(WebKit::PluginView::accessibilityObject const):
(WebKit::PluginView::initializePlugin):
(WebKit::PluginView::layerHostingStrategy const):
(WebKit::PluginView::platformLayer const):
(WebKit::PluginView::graphicsLayer const):
(WebKit::PluginView::scroll):
(WebKit::PluginView::scrollPositionForTesting const):
(WebKit::PluginView::horizontalScrollbar):
(WebKit::PluginView::verticalScrollbar):
(WebKit::PluginView::wantsWheelEvents):
(WebKit::PluginView::paint):
(WebKit::PluginView::countFindMatches):
(WebKit::PluginView::findString):
(WebKit::PluginView::selectionString const):
(WebKit::PluginView::handleEvent):
(WebKit::PluginView::handleEditingCommand):
(WebKit::PluginView::isEditingCommandEnabled):
(WebKit::PluginView::willDetachRenderer):
(WebKit::PluginView::usesAsyncScrolling const):
(WebKit::PluginView::scrollingNodeID const):
(WebKit::PluginView::didAttachScrollingNode):
(WebKit::PluginView::liveResourceData const):
(WebKit::PluginView::performDictionaryLookupAtLocation):
(WebKit::PluginView::notifyWidget):
(WebKit::PluginView::viewGeometryDidChange):
(WebKit::PluginView::viewVisibilityDidChange):
(WebKit::PluginView::clipRectInWindowCoordinates const):
(WebKit::PluginView::focusPluginElement):
(WebKit::PluginView::pendingResourceRequestTimerFired):
(WebKit::PluginView::shouldCreateTransientPaintingSnapshot const):
(WebKit::PluginView::isBeingDestroyed const):
(WebKit::PluginView::pdfDocumentForPrinting const):
(WebKit::PluginView::pdfDocumentSizeForPrinting const):
(WebKit::PluginView::accessibilityHitTest const):
(WebKit::PluginView::lookupTextAtLocation const):
(WebKit::PluginView::rectForSelectionInRootView const):
(WebKit::PluginView::contentScaleFactor const):
(WebKit::PluginView::isUsingUISideCompositing const):
(WebKit::PluginView::didChangeSettings):
(WebKit::PluginView::windowActivityDidChange):
(WebKit::PluginView::didSameDocumentNavigationForFrame):
* Source/WebKit/WebProcess/Plugins/PluginView.h:

Canonical link: https://commits.webkit.org/274794@main
